Update DaemonSet scheduling version inconsistency (#17476)
kubernetes/website: content/en/docs/concepts/workloads/controllers/daemonset.md

{{< feature-state state="beta" for-kubernetes-version="1.12" >}}

> A DaemonSet resource type ensures that all eligible nodes run a copy of a Pod. Normally, the

Reviewer: Why this change? Because DaemonSet is written with titlecase it looks like a type of Kubernetes object, and it is.

Author: Since the previous section only had references to DaemonSet controller, I wanted to be explicit and re-center the conversation to make it clear we were now once again talking about the DaemonSet resource type rather than just the controller, let me know if you prefer it without

Reviewer: Previously Kubernetes talked about “ReplicationController” etc, as if the control loop were more important than the resource. Things are moving to talk about ReplicaSet, Deployment, DaemonSet etc rather than “a ReplicaSet controller”. The main exception to that is if you are actually documenting the specifics of how those control loops are implemented.

node that a Pod runs on is selected by the Kubernetes scheduler. However,

> DaemonSet pods can also be created and scheduled by a DaemonSet controller instead.

Reviewer: How about:
```suggestion
The DaemonSet controller can also schedule the DaemonSet pods that it creates.
```
?

> That introduces the following issues:

Reviewer: This text reads like it's describing a new feature, rather than something that has been the default for several releases.
`ScheduleDaemonSetPods` is the default behavior and, even though it's in beta, to me it makes sense to document the older behavior as the exceptional one.

@christophercarney do you agree, and if so do you want to cover that clean-up in this PR?

Author: I agree and that was something that occurred to me as I was writing so I'm glad you pointed that out. I will make that change in the scope of this PR.
